test/jdk/java/awt/Robot/RobotModifierMaskTest.java

```diff
@@ -0,0 +1,275 @@
+        robot.keyPress(KeyEvent.VK_V);
+        robot.keyRelease(KeyEvent.VK_V);
+        robot.keyRelease(KeyEvent.VK_META);
+
+        robot.delay(100);
+        checkResult(EXPECTED_RESULT_META, "For Command key: ");
+    }
+
+    private static void testAltKey() throws Exception {
+        invokeAndWait(() -> jTextArea.setText(""));
+
+        for (int i = 0; i < 3; ++i) {
+            robot.keyPress(KeyEvent.VK_ALT);
+            robot.keyPress(KeyEvent.VK_A);
+            robot.keyRelease(KeyEvent.VK_A);
+            robot.keyRelease(KeyEvent.VK_ALT);
+            robot.delay(50);
+        }
+
+        robot.delay(100);
+        checkResult(EXPECTED_RESULT_ALT, "For Alt key: ");
+    }
+
+    private static void testCtrlKey() throws Exception {
+        invokeAndWait(() -> jTextArea.setText(""));
+
+        for (int i = 0; i < 5; ++i) {
+            robot.keyPress(KeyEvent.VK_SHIFT);
+            robot.keyPress(KeyEvent.VK_A);
+            robot.keyRelease(KeyEvent.VK_A);
+            robot.keyRelease(KeyEvent.VK_SHIFT);
+            robot.delay(50);
+        }
+
+        robot.delay(50);
+        robot.keyPress(KeyEvent.VK_CONTROL);
+        robot.keyPress(KeyEvent.VK_A);
+        robot.keyRelease(KeyEvent.VK_A);
+        robot.keyRelease(KeyEvent.VK_CONTROL);
+
+        robot.delay(100);
+
+        checkResult(EXPECTED_RESULT_CTRL, "For Control key: ");
+    }
+
+    private static void createInstructionsUI() {
+        jFrame = new JFrame("Manual Test Instructions");
+        jTextArea = new JTextArea(INSTRUCTIONS);
+
+        JScrollPane pane = new JScrollPane(jTextArea);
+        jFrame.getContentPane().add(pane, BorderLayout.CENTER);
+
+        JButton jButton = new JButton("Start");
+        jButton.addActionListener(e -> startTest = true);
+        jFrame.getContentPane().add(jButton, BorderLayout.PAGE_END);
+
+        jFrame.setSize(560,200);
+        jFrame.setLocation(200, 200);
+        jFrame.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);
+        jFrame.setVisible(true);
+    }
+
+    private static void createTestUI() {
+        String mode = isManual ? "MANUAL" : "AUTOMATED";
+        jFrame = new JFrame("RobotModifierMaskTest - Mode: " + mode);
+        jTextArea = new JTextArea("");
+        JScrollPane pane = new JScrollPane(jTextArea);
+        jFrame.getContentPane().add(pane, BorderLayout.CENTER);
+
+        jFrame.setSize(450,100);
+        jFrame.setLocation(200, 200);
+        jFrame.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);
+        jFrame.setVisible(true);
+    }
+
+    private static void checkResult(String expectedResult, String prefixText)
+                                                                throws Exception {
+        invokeAndWait(() -> {
+            boolean condition = expectedResult.equals(EXPECTED_RESULT_CTRL)
+                                ? (jTextArea.getCaretPosition()
+                                     != Integer.parseInt(EXPECTED_RESULT_CTRL))
```

Comment:
You pass different expected result strings but here you always compare `expectedResult` to `EXPECTED_RESULT_CTRL`. Is it intentional? Do I miss anything?

I haven't run the test.

Reply:
Yes, the assertion for CTRL key is different for the rest and involves caret position comparison instead of expected strings (as in other cases), hence I have used `EXPECTED_RESULT_CTRL` to differentiate the cases.

Reply:
@aivanov-jdk Will be restructuring test case using Predicate as suggested.